3rdparty/pstl/CMakeLists.txt

```diff
@@ -1,6 +1,10 @@
-if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang") # only non-multi-generators and clang supported
+if(NBL_BUILD_DPL AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang") # only non-multi-generators and clang ifsupported
```

Comment:
I couldn't get OneApi to build correctly on linux so I just made it an option and disable it when I built this.

Reply:
This is for android btw, we used to checked with "is Clang" for android

Baiscally the oneAPI provides Clang with `par_unseq` for the parallel `for_each`

Reply:
We could replaces the OneAPI with https://github.com/llvm/llvm-project/tree/main/pstl

Reply:
https://lists.llvm.org/pipermail/libcxx-dev/2020-September/000944.html

Reply:
actually there is no problem to go without it on `gcc`, since it supports parallel algorithms and execution policies https://en.cppreference.com/w/cpp/compiler_support/17#C.2B.2B17_library_features

Reply:
Clang 15.0 also seems to have `<execution>` we should nuke the DPL and OneAPI repos from 3rdparty and build system

Android SDK can suck it till it upgrades to Clang 16.0 or whatever.

Reply:
You want me to do that in this pr?

Reply:
@AnastaZIuk will do it